RecoEgamma/EgammaPhotonAlgos/src/ConversionHitChecker.cc

```diff
@@ -4,76 +4,55 @@
 #include "RecoEgamma/EgammaPhotonAlgos/interface/ConversionHitChecker.h"
 // Framework
 //
-#include "TrackingTools/PatternTools/interface/Trajectory.h"
-
-//
 
 
-std::pair<uint8_t,Measurement1DFloat> ConversionHitChecker::nHitsBeforeVtx(const Trajectory &traj, const reco::Vertex &vtx, double sigmaTolerance) const {
+std::pair<uint8_t,Measurement1DFloat> ConversionHitChecker::nHitsBeforeVtx(const reco::TrackExtra & track, const reco::Vertex &vtx, float sigmaTolerance) const {
 
-  const std::vector<TrajectoryMeasurement> &measurements = traj.measurements();
-
-  //protect for empty trajectory
-  if (measurements.size()==0) {
-    return std::pair<unsigned int,Measurement1DFloat>(0,Measurement1DFloat());
-  }
-
-  //check ordering of measurements in trajectory
-  bool inout = (traj.direction() == alongMomentum);
+  // track hits are always inout
```

Comment:
is outside in by construction in reco::Track. many other places rely on this.
there is no easy way to assert it